WalletWasabi/Tor/Control/TorControlClient.cs

```diff
@@ -432,6 +432,10 @@ private async Task ReaderLoopAsync()
 			Logger.LogError($"Exception occurred in the reader loop: {e}.");
 			throw;
 		}
+		finally
+		{
+			SyncChannel.Writer.Complete();
```

Comment:
Maybe close it with an exception if it was there?

Reply:
`Complete` accepts an `Exception` when the stream is closed due to a failure. So, in cases when you don't rethrow it makes sense to store an exception into a variable and pass it to the method, but then the channel reader will throw too.

Reply:
The reader will throw the correct exception instead of reporting that the channel is closed and there's nothing to read, i.e. correct error propagation.

Reply:
Here's what you get when the stream is closed (`s_doneWritingSentinel` is a singleton used to mark a stream as completed without any error):

https://github.com/dotnet/runtime/blob/a904ec31054c89d62188fc5dc6134fdf55f07c82/src/libraries/System.Threading.Channels/src/System/Threading/Channels/ChannelUtilities.cs#L57-L67